pkg/asset/cluster/tfvars.go

```diff
@@ -400,15 +399,31 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		for i, w := range workers {
 			workerConfigs[i] = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*machinev1beta1.GCPMachineProviderSpec)
 		}
+
+		publicZoneName := ""
+		publicZoneProject := ""
 		if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-			publicZone, err := gcpconfig.GetPublicZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.BaseDomain)
-			if err != nil {
-				return errors.Wrapf(err, "failed to get GCP public zone")
+
+			if installConfig.Config.GCP.PublicManagedZone == "" {
```

Comment:
Are we handling the case when installConfig.Config.GCP.PublicManagedZone == "None" ?

Reply:
```
ERROR                                              
ERROR Error: Error retrieving record sets for "None": googleapi: Error 404: The 'parameters.managedZone' resource named 'None' does not exist., notFound 
ERROR                                              
ERROR   with module.dns.google_dns_record_set.api_external[0], 
ERROR   on dns/base.tf line 22, in resource "google_dns_record_set" "api_external": 
ERROR   22: resource "google_dns_record_set" "api_external" { 
ERROR                                              
ERROR failed to fetch Cluster: failed to generate asset "Cluster": failure applying terraform for "cluster" stage: failed to create cluster: failed to apply Terraform: exit status 1 
ERROR                                              
ERROR Error: Error retrieving record sets for "None": googleapi: Error 404: The 'parameters.managedZone' resource named 'None' does not exist., notFound 
ERROR                                              
ERROR   with module.dns.google_dns_record_set.api_external[0], 
ERROR   on dns/base.tf line 22, in resource "google_dns_record_set" "api_external": 
ERROR   22: resource "google_dns_record_set" "api_external" { 
ERROR                                              
ERROR 
```

Reply:
It also appears that using `privateManagedZone: None` didn't affect the terraform as expected. The zone was still created, as were the records.

Reply:
The error above is point to a use for publicManagedZone

```
resource "google_dns_record_set" "api_external" {
  count = var.public_endpoints && var.create_public_zone_records ? 1 : 0

  name         = "api.${var.cluster_domain}."
  type         = "A"
  ttl          = "60"
  managed_zone = var.public_zone_name
  project      = var.public_zone_project
  rrdatas      = [var.api_external_lb_ip]
}
```

This was controlled in tfvars by 

```				
CreatePublicZoneRecords: !strings.EqualFold(installConfig.Config.GCP.PublicManagedZone, gcpconfig.SkipZoneCreationFlag),
```

Which should have controlled the count above and set to 0 since this was false. If count=0 is still attempting to resolve all other variables then this could be an issue.

Reply:
Ah... it's the omitempty on the tfvars... a boolean that is false is omitted. And, since the Terraform variables default to true, they become true in the variable's absence.